pinot-timeseries/pinot-timeseries-planner/src/main/java/org/apache/pinot/tsdb/planner/TimeSeriesQueryEnvironment.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.pinot.tsdb.spi.plan.serde.TimeSeriesPlanSerde;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class TimeSeriesQueryEnvironment {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TimeSeriesQueryEnvironment.class);
+  private final RoutingManager _routingManager;
+  private final TableCache _tableCache;
+  private final Map<String, TimeSeriesLogicalPlanner> _plannerMap = new HashMap<>();
+
+  public TimeSeriesQueryEnvironment(PinotConfiguration config, RoutingManager routingManager, TableCache tableCache) {
+    _routingManager = routingManager;
+    _tableCache = tableCache;
+  }
+
+  public void init(PinotConfiguration config) {
+    String[] languages = config.getProperty(PinotTimeSeriesConfiguration.getEnabledLanguagesConfigKey(), "")
+        .split(",");
+    LOGGER.info("Found {} configured time series languages. List: {}", languages.length, languages);
+    for (String language : languages) {
+      String configPrefix = PinotTimeSeriesConfiguration.getLogicalPlannerConfigKey(language);
+      String klassName =
+          config.getProperty(PinotTimeSeriesConfiguration.getLogicalPlannerConfigKey(language));
+      Preconditions.checkNotNull(klassName, "Logical planner class not found for language: " + language);
+      // Create the planner with empty constructor
+      try {
+        Class<?> klass = TimeSeriesQueryEnvironment.class.getClassLoader().loadClass(klassName);
+        Constructor<?> constructor = klass.getConstructor();
+        TimeSeriesLogicalPlanner planner = (TimeSeriesLogicalPlanner) constructor.newInstance();
+        planner.init(config.subset(configPrefix).toMap());
+        _plannerMap.put(language, planner);
+      } catch (Exception e) {
+        throw new RuntimeException("Failed to instantiate logical planner for language: " + language, e);
+      }
+    }
+    TableScanVisitor.INSTANCE.init(_routingManager);
+  }
+
+  public TimeSeriesLogicalPlanResult buildLogicalPlan(RangeTimeSeriesRequest request) {
+    Preconditions.checkState(_plannerMap.containsKey(request.getLanguage()),
+        "No logical planner found for engine: %s. Available: %s", request.getLanguage(),
+        _plannerMap.keySet());
+    return _plannerMap.get(request.getLanguage()).plan(request);
+  }
+
+  public TimeSeriesDispatchablePlan buildPhysicalPlan(RangeTimeSeriesRequest timeSeriesRequest,
+      RequestContext requestContext, TimeSeriesLogicalPlanResult logicalPlan) {
+    // Step-1: Find tables in the query.
+    final Set<String> tableNames = new HashSet<>();
+    findTableNames(logicalPlan.getPlanNode(), tableNames::add);
+    Preconditions.checkState(tableNames.size() == 1,
+        "Expected exactly one table name in the logical plan, got: %s",
+        tableNames);
+    String tableName = tableNames.iterator().next();
+    // Step-2: Compute routing table assuming all segments are selected. This is to perform the check to reject tables
```

Comment:
To avoid this we will need to make sure in write path all segments for a table go to single server, till we implement multi server in phase2.  Can we capture the limitations of phase1 in task list?